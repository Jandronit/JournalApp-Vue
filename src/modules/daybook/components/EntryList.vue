<template>
<div class="entry-list-container">
  <div class="px-2 pt-2">
    <input type="text"
           class="form-control"
           v-model="term"
           placeholder="Buscar entrada"/>

    <div class="entry-scrollbar">
      <Entry v-for="item in entriesByTerm" :key="item">

      </Entry>
    </div>
  </div>
</div>
</template>

<script>

import {defineAsyncComponent} from 'vue';
import { mapGetters } from 'vuex';

export default {
  components: {
  Entry: defineAsyncComponent(() => import('./Entry.vue'))
    },
  computed: {
    ...mapGetters('journal', ['getEntriesByTerm']),
    entriesByTerm(){
      return this.getEntriesByTerm(this.term);
    }
  },
  data() {
    return {
      term: ''
    }
  }
}
</script>

<style lang="scss" scoped>

.entry-list-container{
  border-right: 1px solid #2c3e50;
  height: calc(100vh - 56px);
}

.entry-scrollbar{
  height: calc(100vh - 110px);
  overflow: scroll;
}
</style>
