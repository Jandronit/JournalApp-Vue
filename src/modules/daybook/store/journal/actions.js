import journalApi from '@/api/journalApi';

// export const myAction = async ({ commit }) => {
//   // do something
// }

export const loadEntries = async ( { commit } ) => {
    const { data } = await journalApi.get('/entries.json')
    if ( !data ) {
        commit('setEntries', []);
        return
    }
    const entries = []
    for ( let id of Object.keys( data ) ) {
        entries.push({
            id,
            ...data[id]
        })
    }
    commit('setEntries', entries)
}

export const updateEntry = async ({ commit }, entry) => {
    const { date, picture, text } = entry
    const dataToSave = { date, picture, text }

    const resp = await journalApi.put(`/entries/${ entry.id }.json`, dataToSave)

    resp.status === 200 && commit('updateEntry', { ...entry })
}

export const createEntry = async ({ commit }, entry) => {
    const { date, picture, text } = entry
    const dataToSave = { date, picture, text }

    const resp = await journalApi.post(`/entries.json`, dataToSave)

    resp.status === 200 && commit('addEntry', { ...dataToSave, id: resp.data.name })

    return resp.data.name
}

export const deleteEntry = async ({ commit }, id) => {
    const resp = await journalApi.delete(`/entries/${ id }.json`)
    resp.status === 200 && commit('deleteEntry', id)

    return id
}
