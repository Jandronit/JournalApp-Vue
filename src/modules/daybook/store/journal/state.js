
export default () =>  ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, quod.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            text: 'Voluptate culpa quia, quod, quas, voluptatum quibusdam quidem voluptates quae quos voluptas quas.',
            picture: null,
        },
        {
            id: new Date().getTime() + 2000,
            date: new Date().toDateString(),
            text: 'Fugiat consequuntur, quod, quas, voluptatum quibusdam quidem voluptates quae quos voluptas quas.',
            picture: null,
        }
    ]
})
