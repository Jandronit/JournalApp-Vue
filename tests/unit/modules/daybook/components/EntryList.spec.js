import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";

import journal from "@/modules/daybook/store/journal";
import EntryList from "@/modules/daybook/components/EntryList.vue";

import { journalState } from "../../../../mocks/test-journal-state";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });
describe("Test in component EntryList", () => {
  const store = createVuexStore(journalState);
  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryList, {
      props: {
        entry: journalState.entries[0],
      },
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test("It must show the getEntriesByTerm without term and show 2 entries", () => {
    expect(wrapper.findAll("entry-stub").length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("It must show the getEntriesByTerm and filter the entries", async () => {
    const input = wrapper.find("input");
    await input.setValue("1");
    expect(wrapper.findAll("entry-stub").length).toBe(1);
  });

  test("Should button new redirect to /new ", () => {
    wrapper.find("button").trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: "new" },
    });
  });
});
