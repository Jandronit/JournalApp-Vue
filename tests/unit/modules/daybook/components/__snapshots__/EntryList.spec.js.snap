// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component EntryList It must show the getEntriesByTerm without term and show 2 entries 1`] = `
<div class="entry-list-container" entry="[object Object]">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fas fa-plus-circle"></i> Nueva Entrada </button></div>
  <div class="entry-scrollbar">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
